aws/nextflow-compute.tf

```diff
@@ -44,39 +46,80 @@ resource "aws_batch_compute_environment" "nf_spot" {
   depends_on   = [aws_iam_role_policy_attachment.nf_batch_role]
 }
 
-# Create an ondemand environment with up to 20 vcpus
-# the AMI used is described in setup-log.md
-resource "aws_batch_compute_environment" "nf_ondemand" {
-  compute_environment_name = "nextflow-ondemand-compute"
+# Create an spot instance0 environment with up to 32 vcpus with large disks
+
+resource "aws_batch_compute_environment" "nf_spot_bigdisk" {
+  compute_environment_name = "nextflow-spot-compute-bigdisk"
   compute_resources {
     instance_role = aws_iam_instance_profile.nf_ecs_instance_role.arn
     instance_type = [
       "optimal",
     ]
-    allocation_strategy = "BEST_FIT"
+    allocation_strategy = "SPOT_CAPACITY_OPTIMIZED"
+    spot_iam_fleet_role = aws_iam_role.nf_spotfleet_role.arn
+    bid_percentage = 100
     max_vcpus = 32
     min_vcpus = 0
-    # ccdl-nextflow-base-v1.1 image
-    image_id = "ami-0a17541ba17115761"
+    # large disk launch template
+    launch_template {
+      launch_template_id = aws_launch_template.nf_lt_bigdisk.id
+    }
     # ec2_key_pair = aws_key_pair.nf_keypair.key_name
     security_group_ids = [
       aws_security_group.nf_security.id,
     ]
     subnets = [
       aws_subnet.nf_subnet.id,
     ]
-    type = "EC2"
+    type = "SPOT"
```

Comment:
With the `bid_percentage = 100` change this is effectively the same thing but with greater flexibility for the future (i.e., you can decrease the `bid_percentage`)?

Reply:
Your documentation answered this question so that's a good sign 😄